lib/icinga/apiactions.cpp

```diff
@@ -123,9 +123,18 @@ Dictionary::Ptr ApiActions::ProcessCheckResult(const ConfigObject::Ptr& object,
 	if (params->Contains("ttl"))
 		cr->SetTtl(HttpUtility::GetLastParameter(params, "ttl"));
 
-	checkable->ProcessCheckResult(cr);
-
-	return ApiActions::CreateResult(200, "Successfully processed check result for object '" + checkable->GetName() + "'.");
+	switch (checkable->ProcessCheckResult(cr)) {
+		case ProcessingResultOk:
+			return ApiActions::CreateResult(200, "Successfully processed check result for object '" + checkable->GetName() + "'.");
+		case ProcessingResultNoCheckResult:
+			return ApiActions::CreateResult(400, "Could not process check result for object '" + checkable->GetName() + "' because no check result was passed.");
+		case ProcessingResultCheckableInactive:
+			return ApiActions::CreateResult(400, "Could not process check result for object '" + checkable->GetName() + "' because the object is inactive.");
```

Comment:
If this happens, it isn't the fault of the client sending something bad. I think most likely this would happen if you send a request while the icinga2 process is starting or stopping, so maybe something like 503 Service Unavailable might be more appropriate here.